platform-sdk/docs/proposals/DP-0x/0X - Tss block signing - Cryptography APIs.md

```diff
@@ -0,0 +1,695 @@
+- **EC (Elliptic Curve)**: `Elliptic` is not elliptic in the sense of an `oval circle`. In the field `Fp`, an `EC` is like a non-connected cloud of points where
+  all points satisfy an equation, and all operations are performed modulo `p`. Some elliptic curves are pairing-friendly.
+- **Bilinear Pairings**: These are mathematical functions used in cryptography to map two elements of different groups (in EC, the group is an elliptic curve) to a single value in another group
+  in a way that preserves specific algebraic properties.
+- **Fields**: Mathematical structures where addition, subtraction, multiplication, and division are defined and behave as expected (excluding division by zero).
+- **Groups**: Sets equipped with an operation (like addition or multiplication) that satisfies certain conditions (closure, associativity, identity element, and inverses).
+- **Share**: Represents a piece of the necessary public/private elements to create signatures. In TSS,
+  a threshold number of shares is needed to produce an aggregate signature that the ledger public key can later verify.
+- **Polynomial Commitment**: It enables verification of evaluations of the polynomial at specific points without revealing the entire polynomial.
+
+### Goals
+- **Usability**: Design user-friendly libraries with a public API that are easy to integrate with other projects, such as consensus node and block node.
+- **EVM support**: Generated signature and public keys should be compatible with EVM precompiled functions
+  so that signature validation can be done on smart contracts without incurring an excessive gas cost.
+- **Security**: Our produced code should be able to pass internal and external security audits.
+- **Flexibility**: Minimize the impact of introducing support for other elliptic curves.
+- **Independent Release**: This library should have its release cycle separate from the platform and be easy for both platform and block node to depend on.
+
+### Non-Goals
+- Implement support for elliptic curve cryptography in Java.
+- Support any system/architecture other than: Windows amd64, Linux amd64 and arm64, and MacOS amd64 and arm64.
+
+## Changes
+
+### Core Behaviors
+
+The proposed TSS solution is based on Groth21.
+
+Groth21 is a non-interactive, publicly verifiable secret-sharing scheme where a dealer can construct a Shamir secret sharing of a field element
+and confidentially yet verifiably distribute shares to multiple receivers.
+It includes a distributed resharing protocol that preserves the public key but creates a fresh secret sharing of the secret key and hands it to a set of receivers,
+which may or may not overlap with the original set of shareholders.
+
+
+#### Overview
+Here, the word participant describes parties producing the aggregate signature and aggregate public key.
+
+Each scheme participant will receive its persistent EC private key from an external source and all participants' EC public keys.
```

Comment:
I think it would make this document a lot clearer to define a term for the persistent EC keypair: `elGamalKeyPair`, `elGamalPrivateKey`, `elGamalPublicKey`

In its current form, it's very easy to get the persistent key confused with the non-persistent share keys

Reply:
I agree that they could be easily mistaken for each other, but calling elGamal to EC keys makes things more complicated to understand.

The document has explained what EC keys are. If we add that they are called Elgamal or EC keys, depending on their function, it would be difficult to remember when reading.